lib/iris/coords.py

```diff
@@ -1762,30 +1761,42 @@ def data(self, data):
         if data is None:
             raise ValueError('The data payload of a CellMeasure may not be '
                              'None; it must be a numpy array or equivalent.')
+        if is_lazy_data(data) and data.dtype.kind in 'iu':
```

Comment:
Let's stretch a crazy thing a little further: do you think we might need to cope with boolean cell measures? We have already discussed (offline) the concept of a cell measure containing only 1 or 0, which is effectively a boolean cell measure...